### ISSUE TITLE
HTML files in directory "tables" are not written

### ISSUE THREAD

**Issue by mapausch** [schemaspy/schemaspy#143]
tried with SchemaSpy6 RC1 and current schemaspy@latest:

The HTML files under <output-directory>/tables are not written. So when I click on a table name then I just get a "file not found".

During run SchemaSpy throws an exception:

```
  ____       _                          ____
 / ___|  ___| |__   ___ _ __ ___   __ _/ ___| _ __  _   _
 \___ \ / __| '_ \ / _ \ '_ ` _ \ / _` \___ \| '_ \| | | |
  ___) | (__| | | |  __/ | | | | | (_| |___) | |_) | |_| |
 |____/ \___|_| |_|\___|_| |_| |_|\__,_|____/| .__/ \__, |
                                             |_|    |___/

                                              6.0.0-rc1

SchemaSpy generates an HTML representation of a database schema's relationships.

INFO  - Starting Main v6.0.0-rc1 on schemaspy with PID 15528 (/srv/www/vhosts/schemaspy/bin/schemaspy/schemaspy-6.0.0-rc1_latest20170830.jar started by root in /srv/www/vhosts/schemaspy/scripts/customer1)
INFO  - No active profile set, falling back to default profiles: default
INFO  - Configuration file not found
INFO  - Starting schema analysis
INFO  - Using database properties:
INFO  -   /srv/www/vhosts/schemaspy/scripts/customer1/./customer1_db_qa_schemaspy6.properties
INFO  - Connected to Oracle - Oracle Database 12c Enterprise Edition Release 12.1.0.2.0 - 64bit Production
With the Partitioning, Automatic Storage Management, OLAP, Advanced Analytics
and Real Application Testing options
INFO  - Gathering schema details
Gathering schema details........................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................(105sec)
Connecting relationships........................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................(116sec)
Writing/graphing summary.INFO  - Gathered schema details in 116 seconds
INFO  - Writing/graphing summary
.........dot -Tpng:cairo ../../schemas/customer1/./customer1_db_qa_schemaspy6/diagrams/orphans/LOGTEXT.1degree.dot -o../../schemas/customer1/./customer1_db_qa_schemaspy6/diagrams/orphans/LOGTEXT.1degree.png -Tcmapx: Warning: cell size too small for content
dot -Tpng:cairo ../../schemas/customer1/./customer1_db_qa_schemaspy6/diagrams/orphans/LOGTEXT.1degree.dot -o../../schemas/customer1/./customer1_db_qa_schemaspy6/diagrams/orphans/LOGTEXT.1degree.png -Tcmapx: in label of node LOGTEXT
.....ERROR - null
java.lang.NullPointerException: null
        at java.io.Writer.write(Writer.java:157)
        at org.schemaspy.view.HtmlRoutinesPage.writeHeader(HtmlRoutinesPage.java:93)
        at org.schemaspy.view.HtmlRoutinesPage.write(HtmlRoutinesPage.java:58)
        at org.schemaspy.SchemaAnalyzer.generateHtmlDoc(SchemaAnalyzer.java:437)
        at org.schemaspy.SchemaAnalyzer.analyze(SchemaAnalyzer.java:251)
        at org.schemaspy.SchemaAnalyzer.analyze(SchemaAnalyzer.java:98)
        at org.schemaspy.Main.runAnalyzer(Main.java:79)
        at org.schemaspy.Main.run(Main.java:72)
        at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:776)
        at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:760)
        at org.springframework.boot.SpringApplication.afterRefresh(SpringApplication.java:747)
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:315)
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1162)
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1151)
        at org.schemaspy.Main.main(Main.java:55)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48)
        at org.springframework.boot.loader.Launcher.launch(Launcher.java:87)
        at org.springframework.boot.loader.Launcher.launch(Launcher.java:50)
        at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:51)
INFO  - Started Main in 834.964 seconds (JVM running for 836.384)
```

I run SchemaSpy with following command:
```
jdk1.8.0_144/bin/java -r schemaspy-6.0.0-rc1_latest20170830.jar -dp ojdbc7_ora12102.jar \
-cat catalog -t customer1_db_qa_schemaspy6.properties -s MYSCHEMA -u MYDBUSER  -p MYDBPASS \
-o schemas/customer1_db_qa_schemaspy6 

```

**Comment by Riggs333** [comment]
The NullPointerException arises because the command line parameter **db** (for database name) is not set.
We need to fix the code and handle that case when it can be null.

But in the meantime could you please try again with setting the db parameter accordingly?

**Comment by mapausch** [comment]
Thanks, this worked :-)

I use Oracle SERVICE_NAME instead of SID for connecting to the database. That's why I didn't use -db on commandline but specified it in the properties-file:
`connectionSpec=jdbc:oracle:thin:@//<host>:<port>/<service_name>`

From my view this issue can be closed.
But feel free to assigne a label and a milestone and keep it open as a "reminder" :-)

**Comment by Riggs333** [comment]
I suggest we fix this with #145.
The class where the NullPointerException was raised (`HtmlRoutinesPage`) needs to be reworked.
